MeshLib/IO/VtkIO/VtuInterface.cpp

```diff
@@ -77,12 +77,17 @@ bool VtuInterface::writeToFile(std::string const &file_name)
     // and PETSC_COMM_WORLD should be replaced with the argument.
     int mpi_rank;
     MPI_Comm_rank(PETSC_COMM_WORLD, &mpi_rank);
-    std::string file_name_base = boost::erase_last_copy(file_name, ".vtu");
+    auto const file_name_base = boost::erase_last_copy(file_name, ".vtu");
+
+    auto const dirname = BaseLib::extractPath(file_name_base);
+    auto basename = BaseLib::extractBaseName(file_name_base);
```

Comment:
I chose the variable names in correspondence with the unix cmdline tools `dirname` and `basename`. So the basename does not contain any leading directories.